pilot/pkg/security/model/authentication.go

```diff
@@ -183,8 +187,10 @@ func ConstructSdsSecretConfig(name, requestedType string) *tls.SdsSecretConfig {
 					},
 				},
 			},
-			InitialFetchTimeout: features.InitialFetchTimeout,
-		},
+			// set the fetch timeout to 0 here because workload certs are
```

Comment:
I think we actually only want this when resourceName == SDSDefaultResourceName or resourceName == SDSRootResourceName. Otherwise they are user provided certs which may or may not exist. So may need to change the function a bit to provide this capability?

Reply:
sure

Reply:
added the check